File: tests/fixtures/codegen/inputs/pointers-expected/queries/items/insertunchecked.go
package items

import (
	"context"
	jsonschemas "example.com/pgsid-validation/generated/jsonschemas"
)

const InsertUncheckedSQL = "INSERT INTO items (payload, unchecked) VALUES ('{\"actor\":1}', $1);"

type InsertUncheckedParams struct {
	Payload *jsonschemas.Event `db:"payload"`
}

func (q *Queries) InsertUnchecked(ctx context.Context, params InsertUncheckedParams) error {
	_, err := q.db.Exec(ctx, InsertUncheckedSQL, params.Payload)
	return err
}

File: tests/fixtures/codegen/inputs/structs-expected/typescript/runtime/queries/items/InsertUnchecked.ts
import type { InsertUncheckedParams } from "../../../types/queries/items/InsertUnchecked.js";
export type { InsertUncheckedParams } from "../../../types/queries/items/InsertUnchecked.js";
import type * as pgsid from "./pgsid/queryable.js";
export const insertUncheckedSql = "INSERT INTO items (payload, unchecked) VALUES ('{\"actor\":1}', $1);";
export async function insertUnchecked(db: pgsid.Queryable, params: InsertUncheckedParams): Promise<void> {
    const jsonInput1 = params["payload"] === null ? null : JSON.stringify(params["payload"]);
    await db.query(insertUncheckedSql, [jsonInput1]);
}
